src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const clones = [];
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
You can make a destructuring to get all the properties you need of action. 
It can help to make your code cleaner